### PR TITLE
OHRM-301 : Renewal feature for expired addons

### PR DESCRIPTION
**Pull request by Bashitha**

### CODE REVIEW

File: dbscript/dbscript-1.sql [hunk 1]
```diff
@@ -2040,6 +2040,7 @@ CREATE TABLE `ohrm_marketplace_addon` (
   `status` VARCHAR(30),
   `version` VARCHAR(100),
   `plugin_name` VARCHAR(255),
+  `type` VARCHAR(30) DEFAULT 'free',
```
Comment (samanthajayasinghe) [inline dbscript/dbscript-1.sql]: What about enum type ?

File: symfony/plugins/orangehrmMarketPlacePlugin/lib/service/MarketplaceService.php [hunk 2]
```diff
@@ -300,4 +318,42 @@ public function addonPrerequisitesVerify($addon)
...
+     */
+    public function MarkExpiredAddons()
+    {
+
+        $paidTypeInstalledAddons = $this->getMarketplaceDao()->getPaidTypeInstalledAddons();
+        $expiredAddonNames = [];
+        if(count($paidTypeInstalledAddons)!=0) {
+            foreach ($paidTypeInstalledAddons as $paidTypeInstalledAddon) {
+                $expiredDate = require_once(__DIR__ . '/../../../' . $paidTypeInstalledAddon['PluginName'] . '/expireDate.php');
+                if (time() < $expiredDate) {
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/lib/service/MarketplaceService.php]: May not work in windows

File: symfony/plugins/orangehrmMarketPlacePlugin/test/model/dao/MarketplaceDaoTest.php [hunk 3]
```diff
@@ -60,4 +60,19 @@ public function testUpdateAddon() {
...
 
+    public function testGetPaidTypeInstalledAddons() {
+
+        $result = $this->marketplaceDao->getPaidTypeInstalledAddons();
+        $this->assertEquals(1, count($result));
+        $this->assertEquals(5, $result[0]['id']);
+    }
+
+    public function testGetAddonById() {
+
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/test/model/dao/MarketplaceDaoTest.php]: extra space

File: symfony/plugins/orangehrmMarketPlacePlugin/test/model/dao/MarketplaceDaoTest.php [hunk 4]
```diff
@@ -60,4 +60,19 @@ public function testUpdateAddon() {
         $this->assertEquals(true, $result);
     }
 
+    public function testGetPaidTypeInstalledAddons() {
+
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/test/model/dao/MarketplaceDaoTest.php]: extra space

File: symfony/plugins/orangehrmMarketPlacePlugin/lib/service/MarketplaceService.php [hunk 5]
```diff
@@ -300,4 +318,41 @@ public function addonPrerequisitesVerify($addon)
...
+    /**
+     * @return array $expiredAddonIds
+     * @throws DaoException
+     */
+    public function getExpiredAddons()
+    {
+        return $this->getMarketplaceDao()->getAddonByStatus(MarketplaceDao::ADDON_STATUS_EXPIRED);
+    }
+
+
```
Comment (Damith88) [inline symfony/plugins/orangehrmMarketPlacePlugin/lib/service/MarketplaceService.php]: please remove extra lines.

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/renewAPIAction.class.php [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            return sfView::NONE;
+        } catch (Exception $e) {
+            Logger::getLogger("orangehrm")->error($e->getCode() . ' : ' . $e->getMessage());
+            Logger::getLogger("orangehrm")->error($e->getTraceAsString());
+            echo json_encode($e->getCode());
+            return sfView::NONE;
+        }
+    }
+
+
```
Comment (Damith88) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/renewAPIAction.class.php]: please remove extra line

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/templates/ohrmAddonsSuccess.php [hunk 7]
```diff
@@ -71,43 +76,73 @@
...
-                                       id="<?php echo 'installButton' . $addon['id']; ?>" value="<?php echo __('Install');?>"
-                                       addid=<?php echo $addon['id'] ?>> <?php }
+                                       id="<?php echo 'installButton' . $addon['id']; ?>"
+                                       value="<?php echo __('Install'); ?>"
+                                       addid=<?php echo $addon['id'] ?>
+                                > <?php }
+                            elseif ($canCreate and ($addon['type'] == 'paid') and in_array($addon['id'], $expiredAddons)) { ?>
+                                <input type="button" name="Submit" class="buttons requestRenewBtn"
+                                       id="<?php echo 'requestRenewButton' . $addon['id']; ?>"
+                                       value="<?php echo __('Req. renew'); ?>"
```
Comment (Damith88) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/templates/ohrmAddonsSuccess.php]: please use proper label for the button.

**Pull request merged**